src/nvim/lua/stdlib.c

```diff
@@ -696,9 +667,6 @@ void nlua_state_add_stdlib(lua_State *const lstate, bool is_thread)
     // str_utfindex
     lua_pushcfunction(lstate, &nlua_str_utfindex);
     lua_setfield(lstate, -2, "str_utfindex");
-    // str_byteindex
-    lua_pushcfunction(lstate, &nlua_str_byteindex);
-    lua_setfield(lstate, -2, "str_byteindex");
```

Comment:
this could be renamed to:

    lua_setfield(lstate, -2, "__str_byteindex");

and then the `vim.str_byteindex` that you defined in `runtime/lua/vim/_editor.lua` can call into this.

Reply:
have pushed a fixup as a seperate commit, will squash once reviewed

Thanks for the advice/guidance